### PR TITLE
[wip] new ubi8 based Dockerfile

### PR DESCRIPTION
**Pull request by hasancelik**
- [ ] test new image at OCP(3&4)/minishift

### CODE REVIEW

**Comment by mesutcelik** [comment]
I was able to build that locally with `docker build` with some slight changes to run locally. 


```
FROM registry.access.redhat.com/ubi8/ubi
MAINTAINER Hazelcast, Inc. Integration Team <info@hazelcast.com>

ENV HZ_HOME /opt/hazelcast/
ENV HZ_CP_MOUNT ${HZ_HOME}/external
ENV LANG en_US.utf8

ENV USER_NAME=hazelcast
ENV USER_UID=10001

ENV HZ_VERSION 3.12.1

ARG HZ_MAVEN_DIR=${HZ_VERSION}
ARG REPOSITORY_URL=https://repository.hazelcast.com
ARG NETTY_VERSION=4.1.32.Final
ARG NETTY_TCNATIVE_VERSION=2.0.20.Final

LABEL name="hazelcast/hazelcast-enterprise-openshift-rhel" \
      vendor="Hazelcast, Inc." \
      version="8.0.0" \
      architecture="x86_64" \
      release="${HZ_VERSION}" \
      url="http://www.hazelcast.com" \
      summary="Hazelcast Openshift Image, certified to RHEL 8" \
      description="Starts a standalone Hazelcast server instance to form a cluster based on kubernetes discovery inside Openshift" \
      io.k8s.description="Starts a standalone Hazelcast server instance to form a cluster based on kubernetes discovery inside Openshift" \
      io.k8s.display-name="Hazelcast" \
      io.openshift.expose-services="5701:tcp" \
      io.openshift.tags="hazelcast,java8,kubernetes,rhel8"

RUN mkdir -p $HZ_HOME
RUN mkdir -p $HZ_CP_MOUNT
WORKDIR $HZ_HOME

ADD start.sh $HZ_HOME/start.sh
ADD stop.sh $HZ_HOME/stop.sh

# Add licenses
ADD licenses /licenses

### Atomic Help File
COPY description.md /tmp/

### Disable subscription-manager plugin to prevent redundant logs
RUN sed -i 's/^enabled=.*/enabled=0/g' /etc/dnf/plugins/subscription-manager.conf

RUN dnf config-manager --disable && \
    dnf update -y  && rm -rf /var/cache/dnf && \
    dnf -y update-minimal --security --sec-severity=Important --sec-severity=Critical --setopt=tsflags=nodocs && \
### Add your package needs to this installation line
    dnf -y --setopt=tsflags=nodocs install java-1.8.0-openjdk-devel apr openssl &> /dev/null && \
### Install go-md2man to help markdown to man conversion
    dnf -y --setopt=tsflags=nodocs install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm &> /dev/null && \
    dnf -y --setopt=tsflags=nodocs install golang-github-cpuguy83-go-md2man &> /dev/null && \
    go-md2man -in /tmp/description.md -out /help.1 && \
    dnf -y remove golang-github-cpuguy83-go-md2man && \
    dnf -y clean all

### add hazelcast enterprise
ADD ${REPOSITORY_URL}/release/com/hazelcast/hazelcast-enterprise-all/${HZ_VERSION}/hazelcast-enterprise-all-${HZ_VERSION}.jar $HZ_HOME

### Adding Logging redirector
ADD https://repo1.maven.org/maven2/org/slf4j/jul-to-slf4j/1.7.12/jul-to-slf4j-1.7.12.jar $HZ_HOME

### Adding JCache
ADD https://repo1.maven.org/maven2/javax/cache/cache-api/1.0.0/cache-api-1.0.0.jar $HZ_HOME

### Adding maven wrapper, downloading Hazelcast Kubernetes discovery plugin and dependencies and cleaning up
COPY mvnw $HZ_HOME/mvnw

### Configure Hazelcast
RUN useradd -l -u $USER_UID -r -g 0 -d $HZ_HOME -s /sbin/nologin -c "${USER_UID} application user" $USER_NAME
RUN chown -R $USER_UID:0 $HZ_HOME $HZ_CP_MOUNT
RUN chmod +x $HZ_HOME/*.sh

### Switch to hazelcast user
USER $USER_UID
RUN cd mvnw && \
    chmod +x mvnw && \
    ./mvnw -f dependency-copy.xml \
    -Dnetty.version=${NETTY_VERSION} \
    -Dnetty-tcnative.version=${NETTY_TCNATIVE_VERSION} \
    dependency:copy-dependencies && \
    cd .. && \
    rm -rf $HZ_HOME/mvnw && \
    rm -rf $HZ_HOME/.m2 && \
    chmod -R +r $HZ_HOME

### Expose port
EXPOSE 5701

### Start hazelcast standalone server.
CMD ["/bin/sh", "-c", "./start.sh"]
```

I also removed some k8s specific parts from start.sh to be able to run locally

```
-if [ "$HAZELCAST_KUBERNETES_NAMESPACE" == "" ]; then
-       HAZELCAST_KUBERNETES_NAMESPACE=$( cat /var/run/secrets/kubernetes.io/serviceaccount/namespace )
-fi
-echo "Kubernetes Namespace: $HAZELCAST_KUBERNETES_NAMESPACE"
-
-if [ "$HAZELCAST_KUBERNETES_SERVICE_DNS" == "$HAZELCAST_KUBERNETES_SERVICE_NAME..svc.$HAZELCAST_KUBERNETES_SERVICE_DOMAIN" ]; then
-       HAZELCAST_KUBERNETES_SERVICE_DNS="$HAZELCAST_KUBERNETES_SERVICE_NAME.$HAZELCAST_KUBERNETES_NAMESPACE.svc.$HAZELCAST_KUBERNETES_SERVICE_DOMAIN"
-fi
-echo "Kubernetes Service DNS: $HAZELCAST_KUBERNETES_SERVICE_DNS"
```

I was able to create 2 member cluster and when I docker exec with one of the container I see RHEL8

```
bash-4.4$ cat /etc/*-release
NAME="Red Hat Enterprise Linux"
VERSION="8.1 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.1"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.1 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.1:GA"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.1
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.1"
Red Hat Enterprise Linux release 8.1 (Ootpa)
Red Hat Enterprise Linux release 8.1 (Ootpa)
```

File: Dockerfile-rhel8 [hunk 1]
```diff
@@ -0,0 +1,94 @@
+FROM registry.access.redhat.com/ubi8
```
Comment (mesutcelik) [inline Dockerfile-rhel8]: There are some registiries listed here different than what is included in this pr.
https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image

I used `registry.access.redhat.com/ubi8/ubi` but I think they are the same as what you used.

I also tried to use ubi-minimal but it failed with the following.
```
sed: can't read /etc/dnf/plugins/subscription-manager.conf: No such file or directory
The command '/bin/sh -c sed -i 's/^enabled=.*/enabled=0/g' /etc/dnf/plugins/subscription-manager.conf' returned a non-zero code: 2
```

**Comment by hasancelik** [comment]
@mesutcelik `Dockerfile` you have shared does not contain `ADD hazelcast.xml $HZ_HOME/hazelcast.xml` line. Could you explain why? I think it is good for general usage current image is very openshift-specific.

>sed: can't read /etc/dnf/plugins/subscription-manager.conf: No such file or directory
The command '/bin/sh -c sed -i 's/^enabled=.*/enabled=0/g' /etc/dnf/plugins/subscription-manager.conf' returned a non-zero code: 2

I encountered same error before multiple times, as far as I can see,  some redhat repos are not avaliable for `ubi-minimal`.

**Comment by mesutcelik** [comment]
I removed hazelcast.xml to be able to run docker image locally. There were k8s specific parameters. We should definitely keep them in the PR.

**Pull request closed without merging**